src/http/api/views/account.posts.view.ts

```diff
@@ -0,0 +1,667 @@
+                            .from('reposts')
+                            .innerJoin('posts', 'posts.id', 'reposts.post_id')
+                            .where('reposts.account_id', accountId),
+                    ])
+                    .orderBy('published_date', 'desc') //Apply sorting at the union level
+                    .as('posts_with_source'),
+            )
+            .innerJoin(
+                'accounts as author_account',
+                'author_account.id',
+                'posts_with_source.author_id',
+            )
+            .leftJoin(
+                'accounts as reposter_account',
+                'reposter_account.id',
+                'posts_with_source.reposter_id',
+            )
+            .leftJoin('likes as current_user_likes', function () {
+                this.on(
+                    'current_user_likes.post_id',
+                    'posts_with_source.id',
+                ).andOnVal(
+                    'current_user_likes.account_id',
+                    '=',
+                    defaultAccountId.toString(),
+                );
+            })
+            .leftJoin('reposts as current_user_reposts', function () {
+                this.on(
+                    'current_user_reposts.post_id',
+                    'posts_with_source.id',
+                ).andOnVal(
+                    'current_user_reposts.account_id',
+                    '=',
+                    defaultAccountId.toString(),
+                );
+            })
+            .modify((query) => {
+                if (cursor) {
+                    query.where(
+                        'posts_with_source.published_date',
+                        '<',
+                        cursor,
+                    );
+                }
+            })
+            .where('posts_with_source.deleted_at', null)
+            .where('posts_with_source.in_reply_to', null)
+            .orderBy('posts_with_source.published_date', 'desc')
+            .limit(limit + 1);
+
+        const results = await query;
+
+        const hasMore = results.length > limit;
+        const paginatedResults = results.slice(0, limit);
+        const lastResult = paginatedResults[paginatedResults.length - 1];
+
+        return {
+            results: paginatedResults.map((result: GetProfileDataResultRow) =>
+                this.mapToPostDTO(result, accountId),
+            ),
+            nextCursor: hasMore ? lastResult.published_date : null,
+        };
+    }
+
+    async getPostsByRemoteLookUp(
+        defaultAccountId: number,
+        defaultAccountApId: URL,
+        handle: string,
+        next: string,
+    ): Promise<Result<AccountPosts, GetPostsError>> {
+        const context = this.fedifyContextFactory.getFedifyContext();
+
+        const documentLoader = await context.getDocumentLoader({
+            handle: 'index',
+        });
+
+        // If the next parameter is not a valid URI, return early
+        if (next !== '' && !isUri(next)) {
```

Comment:
We can use the `URL` type instead of checking strings here

Reply:
Catch here is that next can be an empty string `''` or null and both are totally valid. I used URL type but had to use `new URL('about:blank')` [here](https://github.com/TryGhost/ActivityPub/pull/544/files#diff-5a5f4ba0eac8fe9efad4e5cca046af80781f82b3cd56b9da95bce4b1e8d9dcabR116)  for empty next.